dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal
+
+import com.datadog.android.core.internal.privacy.ConsentProvider
+import com.datadog.android.plugin.DatadogPlugin
+import com.datadog.android.plugin.DatadogPluginConfig
+
+internal abstract class SdkFeature {
+    private var featurePlugins: MutableList<DatadogPlugin> = mutableListOf()
+
+    protected fun registerPlugins(
+        plugins: List<DatadogPlugin>,
+        config: DatadogPluginConfig,
+        trackingConsentProvider: ConsentProvider
+    ) {
+        this.featurePlugins = plugins.toMutableList()
+        plugins.forEach {
+            it.register(config)
+            trackingConsentProvider.registerCallback(it)
+            // just to deliver to current value of the consent inside the plugin
+            it.onConsentUpdated(
+                trackingConsentProvider.getConsent(),
+                trackingConsentProvider.getConsent()
```

Comment:
This leads to initial `PENDING` → `PENDING`, `GRANTED ` → `GRANTED `, `NOT_GRANTED` → `NOT_GRANTED` calls on the public interface, which is confusing (IMO requires at least a documentation comment).

To avoid this, I think we could change this new public interface to:
```kotlin
fun onConsentUpdated(newConsent: TrackingConsent)
```
or:
```kotlin
fun onConsentUpdated(previousConsent: TrackingConsent?, newConsent: TrackingConsent)
```
If needed, each plugin can track the previous consent by its own. The previous consent value is not needed for crash reporter plugin. WDYT?

Reply:
We need also the previous consent in the callback to support the logic in the `ConsentAwareDataWriter`. I understand your concern, on the other hand adding an optional there will add more complexity in the `ConsentAwareDataWriter`. Will try to see if I can avoid it somehow.

Reply:
I see 👌. Indeed, initial state sounds much better and will be more obvious. But then, it will couple plugin initialisation with the SDK. As plugins are defined in separate modules, that may be not easy. Anyway, I understand why it's like this 👍.

Reply:
It is not hard...we already have a config , we just need to add a trackingConsent value inside which is already public.